### PR TITLE
Resizable Panels

### PR DESCRIPTION
**Pull request by manuel3108**
When hovering over the borders of the different Panel's the mouse style will change. Then you can click and drag your mouse in order to resize the Panel

### CODE REVIEW

File: src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    /**
+     * called when the mouse is over the {@link Pane}
+     *
+     * @param event the provided {@link MouseEvent}
+     */
+    private void onMouseMoved(MouseEvent event) {
+        if (!isInDragAndDrop) {
+            if (trackTop && event.getY() < OFFSET) {
```
Comment (Greonyral) [inline src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java]: My first thought looks more complicated than necessary. Looks kind of Rectangle::containsPoint(x, y)
https://docs.oracle.com/javase/8/docs/api/java/awt/geom/Rectangle2D.html
If you don't want the class, implement a own rectangle class. Afterwards the constructor needs to define the rectangle areas which should be tracked. and this mutates to 
```
private void onMouseMoved(MouseEvent event) { 
  boolean capture = false;
   for (Rectangle r : this.trackedAreas) {
    capture |= r.contains(event.getX(), event.getY());
  }
  this.capure = capture;
  if (capture) {
     setCursor(); 
  } else {
    cursor.setDefault();
  }
}

File: src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    private boolean trackBottom = false;
+
+    /**
+     * creates a new resizable wrapper around a {@link Pane}
+     *
+     * @param cursor    object in order to change the cursor
+     * @param positions the locations where resize of the {@link Pane} should be tracked
+     * @param node      the actual {@link Pane} to be resized
+     */
+    public ResizableNodeWrapper(CursorState cursor, DragPosition[] positions, T node) {
```
[the remark this answers is not in the document]
Comment (manuel3108) [reply src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java]: What do you think of affd95293a2d53d7e01ead802d59b12bfa5cf7ca ?

[the remark this answers is not in the document]
Comment (manuel3108) [reply src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java]: fixed in  affd952

File: src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+ * Creates a {@link Pane} which can be resized on the specified positions
+ *
+ * @param <T> needs to be of type {@link Pane}
+ */
+public class ResizableNodeWrapper<T extends Pane> extends NodeWrapper<T> {
+
+    private static final int OFFSET = 10;
+    private final CursorState cursor;
+    private boolean capture = false;
+    private boolean isInDragAndDrop = false;
```
[the remark this answers is not in the document]
Comment (manuel3108) [reply src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java]: fixed in 87cfbd3

File: src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        super.getNode().setOnMouseReleased(this::dragLeave);
+    }
+
+    /**
+     * called when the mouse is over the {@link Pane}
+     *
+     * @param event the provided {@link MouseEvent}
+     */
+    private void onMouseMoved(MouseEvent event) {
+        if (!isInDragAndDrop) {
```
[the remark this answers is not in the document]
Comment (manuel3108) [reply src/main/java/org/gleisbelegung/ui/lib/style/ResizableNodeWrapper.java]: fixed in 87cfbd3

File: src/main/java/org/gleisbelegung/ui/main/TrainInformationPanel.java [hunk 5]
```diff
@@ -15,32 +13,48 @@
...
+        //I have no idea why this is necessary here and not on the other classes which do the same
+        Runnable r = () -> {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            pane.addListener(ResizableNodeWrapper.DragPosition.LEFT);
+        };
+        new Thread(r).start();
```
Comment (Greonyral) [inline src/main/java/org/gleisbelegung/ui/main/TrainInformationPanel.java]: You create a new Thread, then start it, leaving the method and thus creating a zombie process (never [joining](https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#join--) the thread).

File: src/main/java/org/gleisbelegung/ui/main/TrainInformationPanel.java [hunk 6]
```diff
@@ -15,32 +13,48 @@
...
         pane.addStyle(new BackgroundColor("#0f0"));
         return pane.getNode();
     }
 
-    @Override public void setSizes() {
+    @Override
+    public void setSizes() {
         pane.getNode().setPrefWidth(200);
+
+        //I have no idea why this is necessary here and not on the other classes which do the same
```
Comment (Greonyral) [inline src/main/java/org/gleisbelegung/ui/main/TrainInformationPanel.java]: Looks like a timing issue - I think you are with another thread in the process of setting things up / initializing. You should synchronize the method caller to call the method as long as necessary not the callee.

**Pull request merged**